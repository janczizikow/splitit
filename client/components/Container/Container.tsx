import styled from "@emotion/styled";

interface Props {
  fluid?: boolean;
}

const Container = styled.div`
  padding: 0 1rem;
  margin: 0 auto;
  max-width: ${(p: Props) => (p.fluid ? "100%" : "1172px")};
`;

Container.displayName = "Container";

export default Container;
