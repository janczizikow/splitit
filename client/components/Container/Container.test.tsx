import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import Container from "./Container";

describe("<Container />", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<Container />);
  });

  it("matches snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("changes styles with fluid prop", () => {
    wrapper.setProps({ fluid: true });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
