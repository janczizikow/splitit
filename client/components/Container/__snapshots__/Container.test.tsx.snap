// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> changes styles with fluid prop 1`] = `
.emotion-0 {
  padding: 0 1rem;
  margin: 0 auto;
  max-width: 100%;
}

<Container
  fluid={true}
>
  <div
    className="emotion-0"
  />
</Container>
`;

exports[`<Container /> matches snapshot 1`] = `
.emotion-0 {
  padding: 0 1rem;
  margin: 0 auto;
  max-width: 1172px;
}

<Container>
  <div
    className="emotion-0"
  />
</Container>
`;
